x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -165,6 +165,11 @@ export class MapEmbeddable extends Embeddable {
     });
   }
 
+  setLayerList(layerList) {
```

Comment:
Consider making this method `async` and returning this._store.dispatch. It will allow clients to wait until the action has resolved (although this _does_ not include awaiting until mapbox actually has rendered everything, which is more complicated).